sound: rework module

normalcea opened the pull request:
Removes the `asahi-audio` configuration options alongside enabling pipewire. This makes `setupAsahiSound` only set up the DSP components and options required rather than enabling sound modules itself.

Should resolve: https://github.com/nix-community/nixos-apple-silicon/issues/352 which explains that the previous implementation is not required anymore. I've tested this for a couple of days on my MBA 2020 and the sound issues with the volume being set to 100% and the GNOME microphone indicator always being present are now gone.

Haven't noticed any notable regressions but I will update this PR if I find them. Also related, we might not need the `pkgs.asahi-audio` package at all if this is all we need and we can retire it from nixpkgs.

normalcea commented:
I'm not sure if `services.pipewire.wireplumber.extraLv2Packages` should also be set? It seems like it should based on the fact that `asahi-audio` was applied to `configPackages` for both wireplumber and pipewire.

dramforever reviewed (changes requested):
This can't be right...

This just removes `pkgs.asahi-audio`, which gets rid of the DSP pipelines (incl. filters) from pipewire. `services.pipewire.extraLv2Packages` also doesn't do anything.

I guess you don't need asahi-audio to hear normal-volume audio. We could make the `speakersafetyd` part and the `$ALSA_CONFIG_UCM2` not conditional on `setupAsahiSound`, but just straight up removing asahi-audio is definitely not right.

normalcea commented:
Okay, I've changed the proposal to remove the sound module itself and transition it into `hardware.asahi.enable` so that the sound components required for asahi are conditionally added if sound modules (in this case, pipewire) are enabled.

It also changed pulseaudio to `lib.mkForce` since we don't want users to be using pulseaudio if at all. 

I'm also not sure about users just using ALSA and whether speakersafetyd should be activated for them as well.

sezaru commented:
I don't have the link anymore, but when I was researching about the issue, I do remember reading somewhere that adding the configs to pipewire and wireplumber were not needed anymore and it was fully removed from Fedora Asahi.

That's is also why asahi-diagnose complains with the default nix config.
